pyproject.toml

```diff
@@ -17,11 +18,35 @@ line-length = 120
 target-version = "py38"
 # See Rules in Ruff documentation: https://beta.ruff.rs/docs/rules/
 select = [
+    "B",        # bugbear
+    "E",        # pycodestyle
+    "F",        # pyflakes
+    "INP",      # flake8-tidy-imports
+    "W",        # pycodestyle
     "I",        # isort
-    "F401",     # Unused imports
     "UP",       # pyupgrade
     "TID251",   # Disallowed imports (flake8-tidy-imports.banned-api)
+    "PYI",      # flake8-pyi
+    "RUF100",   # Equivalent to flake8-noqa NQA103
+    "PGH004",   # Equivalent to flake8-noqa NQA104
 ]
+ignore = ["PYI021", "PYI024", "PYI041", "PYI043"]
+
+[tool.ruff.per-file-ignores]
+"*.pyi" = [
+    "B",
+    "E501",
+    "E741",
+    "E743",
+    "F401", # Unused import
```

Comment:
I'm not sure, but I don't think we should completely ignore `F401`.

As that wouldn't clear up places where there's no re-export and an import becomes unused.